pkg/driver/controller.go

```diff
@@ -170,10 +174,41 @@ func parseVolParams(params map[string]string) (common.HSVolumeParameters, error)
 	return vParams, nil
 }
 
-func (d *CSIDriver) ensureShareBackedVolumeExists(_ context.Context, hsVolume *common.HSVolume) error {
+func (d *CSIDriver) ensureNFSDirectoryExists(ctx context.Context, backingShareName string, hsVolume *common.HSVolume) error {
```

Comment:
This function is helper function for a code where we wanted to put all nfs dir inside one base share.

It will only come in play when user provide eg mountbackingshare: share1, in storage class.
Note it will only affect the pvc created newly after the change not the old created one.

So with this we will solve the problem creating 1000 of shares for each dynamically created pvc volumes.

Each pvc will now point to a dir and that will be inside a base share.
And if user don't want that then he can choose the old way of doing it creating a new share for each new pvc.